File: examples/indexer/javascript/accountsApplicationIDPaging.js
// requires algosdk@1.7 or higher 
// verify installed version
// npm list algosdk

const algosdk = require('algosdk');

// const indexer_server = "https://testnet-algorand.api.purestake.io/idx2/";
// const indexer_port = "";
// const indexer_token = {
//     'X-API-key': 'B3SU4KcVKi94Jap2VXkK83xx38bsv95K5UZm2lab',
// }
const indexer_server = "http://localhost";
const indexer_port = 59998;
const indexer_token = "";
 
// Instantiate the indexer client wrapper
let indexerClient = new algosdk.Indexer(indexer_token, indexer_server, indexer_port);

let nexttoken = "";
let numAccounts = 1;
(async () => {

    let applicationID = 70;
    let limit = 2;
    while (numAccounts > 0) {
        // execute code as long as condition is true
        let next_page = nexttoken;
        let accountInfo = await indexerClient.searchAccounts()
            .applicationID(applicationID).limit(limit).nextToken(next_page).do();
        let accounts = accountInfo['accounts'];
        numAccounts = accounts.length;
        if (numAccounts > 0) {
            nexttoken = accountInfo['next-token'];
            console.log("Information for account info for Application ID: " + JSON.stringify(accountInfo, undefined, 2));
        }
    }

}) ().catch(e => {
    console.log(e);
    console.trace();
});

File: examples/indexer/python/accounts_application_id_paging.py
import json
# requires Python SDK version 1.3 or higher
from algosdk.v2client import indexer

# instantiate indexer client

# myindexer_token = 'B3SU4KcVKi94Jap2VXkK83xx38bsv95K5UZm2lab'
# myindexer_address = 'https://testnet-algorand.api.purestake.io/idx2/'
# myindexer_header = {'X-Api-key': myindexer_token}

myindexer_address = 'http://localhost:59998'

myindexer = indexer.IndexerClient(
    indexer_token="", indexer_address=myindexer_address)
# myindexer = indexer.IndexerClient(
#     indexer_token="", indexer_address=myindexer_address, headers=myindexer_header)

nexttoken = ""
num_accounts = 1
# loop using next_page to paginate until there are no more accounts
# in the response
# (max is 100 default
# unless limit is used for max 1000 per request on accounts)
while (num_accounts > 0):
    response = myindexer.accounts(
        application_id=70, limit=2,next_page=nexttoken)
    accounts = response['accounts']
    num_accounts = len(accounts)
    if (num_accounts > 0):
        nexttoken = response['next-token']
        # Pretty Printing JSON string
        print("Account Info for Application ID: " + json.dumps(response, indent=2, sort_keys=True))
